### ISSUE TITLE
Add missing type hints

### ISSUE THREAD

**Issue by alifeee** [burnash/gspread#1321]
This work should be done in the [6.0.0 release branch](https://github.com/burnash/gspread/tree/feature/release_6_0_0/)

Run `mypy --install-types --non-interactive --ignore-missing-imports ./gspread` (from CI) to see which types are broken/missing

**Comment by bpshaver** [comment]
Looking into this.

`mypy` doesn't like that the named tuple singleton instances are created at the same time as the named tuple class is created.

```python
Dimension = namedtuple("Dimension", ["rows", "cols"])("ROWS", "COLUMNS")
```

It seems like a better solution for this is a `StrEnum`. I'll see about making the change and open a PR.

**Comment by alifeee** [comment]
Hi! Thanks for looking into it!

One thing I forgot to mention in the issue: this work should be done on the [6.0.0 release branch](https://github.com/burnash/gspread/tree/feature/release_6_0_0/). On this branch, as you suggested, we have already [switched to StrEnums](https://github.com/burnash/gspread/pull/1250).

Also, see the linked PR #1337. I (think that I) fixed the remaining type issues there, and the work is complete. If you are still interested, do have a look. Some things may be patched over with `Any` that could be done differently, etc.

**Comment by bpshaver** [comment]
Well drats. I've already made all the necessary changes and was about to open a PR. Didn't see anything in my skim of the contributing guide about checking out the latest release branch, although that does make sense.

**Comment by bpshaver** [comment]
Consider [this](https://github.com/burnash/gspread/pull/1339) a suggestion. You can only install `strenum` if the python version is below 3.11.

**Comment by alifeee** [comment]
closed by #1337

> Well drats. I've already made all the necessary changes and was about to open a PR. Serves me right for skimming this issue.

My bad for not including the branch in the issue description. I should have done this, especially as I tagged this as `good first issue`
